src/controller/CHIPDeviceController.cpp

```diff
@@ -2522,6 +2562,43 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
     }
 }
 
+void DeviceCommissioner::ExtendFailsafeBeforeNetworkEnable(DeviceProxy * device, CommissioningParameters & params,
+                                                           CommissioningStage step)
+{
+    auto * commissioneeDevice = FindCommissioneeDevice(device->GetDeviceId());
+    if (device != commissioneeDevice)
+    {
+        // Not a commissionee device; just return.
+        ChipLogError(Controller, "Trying to extend fail-safe for an uknown commissionee with device id " ChipLogFormatX64,
+                     ChipLogValueX64(device->GetDeviceId()));
+        CommissioningStageComplete(CHIP_ERROR_INCORRECT_STATE, CommissioningDelegate::CommissioningReport());
+        return;
+    }
+
+    // Try to make sure we have at least enough time for our expected
+    // commissioning bits plus the MRP retries for a Sigma1.
+    uint16_t failSafeTimeoutSecs = params.GetFailsafeTimerSeconds().ValueOr(kDefaultFailsafeTimeout);
```

Comment:
There is a specific value in NetworkCommissioning cluster that would be the right value: `ConnectMaxTimeSeconds`

Reply:
Let me look into doing that.

Reply:
The main issue: That time just captures the connect time, not the time it takes to receive the response or the time it takes to then do operational discovery, a CASE handshake, etc.

That's why I'm using the device's estimate of total time it would need a fail-safe for instead, but adding some padding for the MRP bits to it.  If we used the ConnectMaxTimeSeconds, in addition to threading it through to here (which is doable), we would also have to add various overheads on to it...